format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatProvider.java

```diff
@@ -115,8 +122,23 @@ public Schema detectSchema(FormatContext context, InputFiles inputFiles) throws
         Sheet workSheet;
         // Check if user wants to access with name or number
         if (conf.getSheet() != null && conf.getSheet().equals(XlsInputFormatConfig.SHEET_NUMBER)) {
-          workSheet = workbook.getSheetAt(Integer.parseInt(Objects.requireNonNull(conf.getSheetValue())));
+          int sheetValue = 0;
+          if (!Strings.isNullOrEmpty(conf.getSheetValue())) {
+            try {
+              sheetValue = Integer.parseInt(conf.getSheetValue());
+            } catch (NumberFormatException e) {
+              failureCollector.addFailure("Sheet number must be a number.", null)
```

Comment:
this is code is duplicated in the validate() method, it would be better to have a method like getSheetAsNumber() that is called in both places.